### ISSUE TITLE
Questions about CLIP embeddings

### ISSUE THREAD

**Issue by dk-hong** [facebookresearch/Detic#55]
Hi, I have some questions about using CLIP embeddings as classification weights

1. I think there is no other code which load CLIP embeddings as classification weights except load model.
Aren’t the provided weights trained? (Are CLIP embeddings loaded only?)

2. I think it may work if FC layer of a classification network is used as classification weights of detector.
Why are you choose to use CLIP embedding instead of pre-trained classification network?

Thank you.

**Comment by dk-hong** [comment]
I have one more question.
In the abstract of the paper, there is a quote “it generalizes to new datasets without finetuning”.
However, in the loss function, there is a loss term for classification data.
What means “without finetuning?”
- the region proposal network is fixed and doesn’t have to be finetuned (classification weights from CLIP embeddings should be trained)
- classification weights from CLIP embeddings are fixed and not finetuned (other parameters in the detector should be trained)
- after training with the detection dataset and classification dataset, any other classes can be added by attaching CLIP embeddings to classification weights without finetuning
If every description is wrong, I would appreciate it if you explain what it means.

**Comment by xingyizhou** [comment]
Hi,

  Thank you for your interest. 
  1. The CLIP embeddings are pre-computed and dumped to the disk [here](https://github.com/facebookresearch/Detic/blob/main/datasets/metadata/lvis_v1_clip_a%2Bcname.npy). They are loaded by the model [here](https://github.com/facebookresearch/Detic/blob/main/detic/modeling/roi_heads/zero_shot_classifier.py#L40).
  2. Yes it can work. We tried this earlier during the development but it gave worse results than random initialization as far as I remembered. Feel free to try this again yourself.
  3. The region proposal network is fixed ... --> correct.
  4. Classification weights from CLIP embeddings are fixed and not fine-tuned ... --> this is not fully accurate. When testing on a new dataset, we remove the CLIP embeddings of the training dataset and compute a new CLIP embedding using the test vocabulary.
  5. After training with the detection dataset ... --> correct.
